src/test/groovy/org/prebid/server/functional/tests/GeoSpec.groovy

```diff
@@ -0,0 +1,169 @@
+        }
+        def defaultPbsService = pbsServiceFactory.getService(GEO_LOCATION +
+                ["settings.default-account-config": encode(config),
+                 "geolocation.enabled"            : hostGeolocation])
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            device = new Device(
+                    ip: "209.232.44.21",
+                    ipv6: "af47:892b:3e98:b49a:a747:bda4:a6c8:aee2",
+                    geo: new Geo(
+                            country: null,
+                            region: null,
+                            lat: PBSUtils.getRandomDecimal(0, 90),
+                            lon: PBSUtils.getRandomDecimal(0, 90)))
+            ext.prebid.trace = VERBOSE
+        }
+
+        and: "Account in the DB"
+        def accountConfig = new AccountConfig(
+                auction: new AccountAuctionConfig(debugAllow: true),
+                settings: new AccountSetting(geoLookup: accountGeoLookup))
+        def account = new Account(status: ACTIVE, uuid: bidRequest.site.publisher.id, config: accountConfig)
+        accountDao.save(account)
+
+        when: "PBS processes auction request"
+        defaultPbsService.sendAuctionRequest(bidRequest, ["X-Forwarded-For": "209.232.44.21"])
+
+        then: "Bidder request shouldn't contain country and region"
+        def bidderRequests = bidder.getBidderRequest(bidRequest.id)
+        assert !bidderRequests.device.geo.country
+        assert !bidderRequests.device.geo.region
+
+        and: "Metrics processed across geo location shouldn't be updated"
+        def metrics = defaultPbsService.sendCollectedMetricsRequest()
+        assert !metrics["geolocation_requests"]
+        assert !metrics["geolocation_successful"]
+
+        where:
+        defaultAccountGeoLookupConfig | hostGeolocation | accountGeoLookup
+        true                          | "true"          | false
+        true                          | "false"         | true
+        false                         | "false"         | false
+        false                         | "true"          | false
+    }
+
+    def "PBS shouldn't populate geo with country, region and emit error in log and metric when geo look up failed"() {
+        given: "Test start time"
+        def startTime = Instant.now()
+
+        and: "Default basic generic BidRequest"
+        def defaultPbsService = pbsServiceFactory.getService(GEO_LOCATION +
+                ["geolocation.configurations.[0].address-pattern": "302.",
+                 "geolocation.enabled"                           : "true"])
+        def bidRequest = BidRequest.defaultBidRequest.tap {
+            device = new Device(
+                    ip: "209.232.44.21",
+                    ipv6: "af47:892b:3e98:b49a:a747:bda4:a6c8:aee2",
+                    geo: new Geo(
+                            country: null,
+                            region: null,
```

Comment:
what if only country or region is specified? And what if the specified country and region don't match?

Reply:
We don't care, about matching between country and region